src/Algebra/Morphism/MagmaMonomorphism.agda

```diff
@@ -0,0 +1,125 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Consequences of a monomorphism between magma-like structures
+------------------------------------------------------------------------
+
+-- See Data.Nat.Binary.Properties for examples of how this and similar
+-- modules can be used to easily translate properties between types.
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Algebra.Core
+open import Algebra.Structures
+open import Algebra.Morphism.Structures
+open import Relation.Binary.Core
+
+module Algebra.Morphism.MagmaMonomorphism
+  {a b} {A : Set a} {B : Set b}
+  {ℓ₁ ℓ₂} {_≈₁_ : Rel A ℓ₁} {_≈₂_ : Rel B ℓ₂}
+  {_∙_ : Op₂ A} {_◦_ : Op₂ B} {f : A → B}
+  (isMagmaMonomorphism : IsMagmaMonomorphism _≈₁_ _≈₂_ _∙_ _◦_ f)
+  (◦-isMagma : IsMagma _≈₂_ _◦_)
```

Comment:
This argument is redundant for the structures `isSemigroup`, `isBand`, etc. below. Perhaps it could be added just for the properties section via an anonymous module, but not for the structures. The same applies for monoids.

Reply:
Fair enough. I was just trying to make my life easier for myself, but you're right we shouldn't have it everywhere if we don't need it!